<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RulesDialog</class>
 <widget class="QDialog" name="RulesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>651</width>
    <height>543</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Rule</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="2" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Node</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="nodesCombo"/>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QCheckBox" name="nodeApplyAllCheck">
       <property name="text">
        <string>Apply rule to all nodes</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="11" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <spacer name="horizontalSpacer_6">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Apply|QDialogButtonBox::Close|QDialogButtonBox::Help|QDialogButtonBox::Reset</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="10" column="1">
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="8" column="1">
    <widget class="QGroupBox" name="enableGroup">
     <property name="flat">
      <bool>false</bool>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0,0,0,0,0">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <property name="verticalSpacing">
       <number>12</number>
      </property>
      <item row="11" column="1">
       <widget class="QCheckBox" name="dstIPCheck">
        <property name="text">
         <string>To this IP / Network</string>
        </property>
       </widget>
      </item>
      <item row="10" column="4" colspan="2">
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="6" column="4" colspan="3">
       <widget class="QLineEdit" name="procLine">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="placeholderText">
         <string>/path/to/executable, .*/bin/executable[0-9\.]+$, ...</string>
        </property>
       </widget>
      </item>
      <item row="8" column="4" colspan="2">
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Action</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QCheckBox" name="dstPortCheck">
        <property name="text">
         <string>To this port</string>
        </property>
       </widget>
      </item>
      <item row="13" column="1">
       <widget class="QCheckBox" name="dstListsCheck">
        <property name="text">
         <string>To this list of domains</string>
        </property>
       </widget>
      </item>
      <item row="4" column="4">
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="11" column="4" colspan="3">
       <widget class="QComboBox" name="dstIPCombo">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>You can specify a single IP:
- 192.168.1.1

or a regular expression:
- 192\.168\.1\.[0-9]+

multiple IPs:
- ^(192\.168\.1\.1|172\.16\.0\.1)$

You can also specify a subnet:
- 192.168.1.0/24

Note: Commas or spaces are not allowed to separate IPs or networks.</string>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
        <item>
         <property name="text">
          <string>LAN</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>127.0.0.0/8</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>192.168.0.0/24</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>192.168.1.0/24</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>192.168.2.0/24</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>192.168.0.0/16</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>169.254.0.0/16</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>172.16.0.0/12</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>10.0.0.0/8</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>::1/128</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>fc00::/7</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>ff00::/8</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>fe80::/10</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>fd00::/8</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="4">
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="10" column="6">
       <widget class="QLineEdit" name="uidLine">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="8" column="6">
       <widget class="QLineEdit" name="dstPortLine">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;You can specify multiple ports using regular expressions:&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;- 53, 80 or 443:&lt;/p&gt;&lt;p&gt;^(53|80|443)$&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;- 53, 443 or 5551, 5552, 5553, etc:&lt;/p&gt;&lt;p&gt;^(53|443|555[0-9])$&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="4" column="6">
       <widget class="QComboBox" name="durationCombo">
        <item>
         <property name="text">
          <string>once</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>30s</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>5m</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>15m</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>30m</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>1h</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>until reboot</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>always</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="12" column="4" colspan="3">
       <widget class="QLineEdit" name="dstHostLine">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Commas or spaces are not allowed to specify multiple domains. 

Use regular expressions instead: 
.*(opensnitch|duckduckgo).com
.*\.google.com

or a single domain:
www.gnu.org - it'll only match www.gnu.org, nor ftp.gnu.org, nor www2.gnu.org, ...
gnu.org         - it'll only match gnu.org, nor www.gnu.org, nor ftp.gnu.org, ...</string>
        </property>
        <property name="placeholderText">
         <string>www.domain.org, .*\.domain.org</string>
        </property>
       </widget>
      </item>
      <item row="12" column="1">
       <widget class="QCheckBox" name="dstHostCheck">
        <property name="text">
         <string>To this host</string>
        </property>
       </widget>
      </item>
      <item row="7" column="4" colspan="3">
       <widget class="QLineEdit" name="cmdlineLine">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Duration</string>
        </property>
       </widget>
      </item>
      <item row="5" column="6">
       <widget class="QComboBox" name="protoCombo">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Only TCP, UDP or UDPLITE are allowed&lt;/p&gt;&lt;p&gt;You can use regexp, i.e.: ^(TCP|UDP)$&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
        <property name="currentText">
         <string>TCP</string>
        </property>
        <item>
         <property name="text">
          <string>TCP</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>UDP</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>UDPLITE</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>TCP6</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>UDP6</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>UDPLITE6</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QCheckBox" name="protoCheck">
        <property name="text">
         <string>Protocol</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QCheckBox" name="procCheck">
        <property name="text">
         <string>From this executable</string>
        </property>
       </widget>
      </item>
      <item row="3" column="6">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QRadioButton" name="actionDenyRadio">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Deny</string>
          </property>
          <property name="icon">
           <iconset theme="emblem-important">
            <normaloff>../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../.designer/backup</normaloff>../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../.designer/backup</iconset>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="actionAllowRadio">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="text">
           <string>Allow</string>
          </property>
          <property name="icon">
           <iconset theme="emblem-default">
            <normaloff>../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../.designer/backup</normaloff>../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../../.designer/backup</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="7" column="1">
       <widget class="QCheckBox" name="cmdlineCheck">
        <property name="text">
         <string>From this command line</string>
        </property>
       </widget>
      </item>
      <item row="10" column="1">
       <widget class="QCheckBox" name="uidCheck">
        <property name="text">
         <string>From this user ID</string>
        </property>
       </widget>
      </item>
      <item row="13" column="4" colspan="3">
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QPushButton" name="selectListButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset theme="system-search"/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="dstListsLine">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select a directory with lists of domains to block or allow.&lt;/p&gt;&lt;p&gt;Put inside that directory files with any extension containing lists of domains.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;The format of each entry of a list is as follow (hosts format):&lt;/p&gt;&lt;p&gt;127.0.0.1 www.domain.com&lt;/p&gt;&lt;p&gt;or &lt;/p&gt;&lt;p&gt;0.0.0.0 www.domain.com&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Name</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QCheckBox" name="enableCheck">
       <property name="text">
        <string>Enable</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_7">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="9" column="1">
    <widget class="QLabel" name="statusLabel">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLineEdit" name="ruleNameEdit">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="toolTip">
      <string>The rules are checked in alphabetical order, so you can name them accordingly to prioritize them.

000-allow-localhost
001-deny-broadcast
...</string>
     </property>
     <property name="placeholderText">
      <string>leave blank to autocreate</string>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QCheckBox" name="precedenceCheck">
       <property name="toolTip">
        <string>If checked, this rule will take precedence over the rest of the rules. No others rules will be checked after this one.

You must name the rule in such manner that it'll be checked first, because they're checked in alphabetical order. For example:

[x] Priority - 000-priority-rule
[  ] Priority - 001-less-priority-rule</string>
       </property>
       <property name="text">
        <string>Priority rule</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_8">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QCheckBox" name="sensitiveCheck">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;By default, the field of the rules are case-insensitive, i.e., if a process tries to access gOOgle.CoM and you have a rule to Deny .*google.com, the connection will be blocked.&lt;br/&gt;&lt;/p&gt;&lt;p&gt;If you check this box, you have to specify the exact string (domain, executable, command line) that you want to filter.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>Case-sensitive</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
